### PR TITLE
[fix] release.sh: pin gh to origin via GH_REPO on multi-remote clones

### PR DESCRIPTION
**Pull request by lugassawan**
## Summary
- Export `GH_REPO` from the `origin` remote URL at script startup so every `gh` invocation is pinned to the correct repo, eliminating "No default remote repository has been set" failures on multi-remote clones (e.g. after `gh repo fork` adds sibling remotes).
- Remove the trailing `gh repo view` call in the Done output; reuse `$GH_REPO` directly.
- Add `tests/test_release_sh_gh_repo.py` with three assertions: export ordering, no orphan `gh` calls above the export, and a dynamic extraction round-trip for all four origin URL forms.

## Test Plan

- [x] Changed skills/commands/agents load without errors
- [ ] Examples in the diff were exercised manually
- [x] `pytest tests/test_release_sh_gh_repo.py -v` → 6 passed
- [x] `pytest tests/ -q` → 509 passed (all green)
- [x] `bash scripts/validate.sh` → All checks passed
- [x] `bash -n scripts/release.sh` → no output (syntax clean)

### Type-tailored checks ([fix])
- [ ] On a multi-remote clone, `./scripts/release.sh patch` no longer aborts at `gh pr merge` with "No default remote repository has been set"
- [ ] `GH_REPO` resolves correctly for SSH (`git@github.com:owner/repo.git`) and HTTPS (`https://github.com/owner/repo.git`) origin URLs — covered by round-trip tests

Issue: N/A — urgent release-script fix

### CODE REVIEW

File: tests/test_release_sh_gh_repo.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        printf '%s\\n' "$GH_REPO"
+    """)
+
+    def _run(self, fake_url: str) -> str:
+        """Stub `git` on PATH to return fake_url, then run the extraction snippet."""
+        stub_dir = Path(os.environ.get("TMPDIR", "/tmp")) / "gh_repo_stub"
+        stub_dir.mkdir(exist_ok=True)
+        stub = stub_dir / "git"
+        stub.write_text(
+            f"#!/bin/sh\necho '{fake_url}'\n"
```
Comment (lugassawan) [inline tests/test_release_sh_gh_repo.py]: **Severity: High — Shell injection via unsanitised `fake_url` in stub**

`fake_url` is interpolated directly between single quotes in:
```python
stub.write_text(f"#!/bin/sh\necho '{fake_url}'\n")
```
Any URL containing a single quote terminates the shell literal, allowing arbitrary shell code to execute with the test runner's privileges.

Suggested fix — use `shlex.quote` (or `printf '%s\n'` in the shell stub):
```python
import shlex
stub.write_text(
    '#!/bin/sh\n'
    f'printf "%s\n" {shlex.quote(fake_url)}\n'
)
```

File: tests/test_release_sh_gh_repo.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            (i for i, ln in enumerate(lines) if "export GH_REPO" in ln),
+            None,
+        )
+        assert export_idx is not None, "export GH_REPO not found in release.sh"
+
+        gh_calls = _gh_call_lines(lines)
+        assert gh_calls, "No non-comment 'gh pr' or 'gh repo' call found in release.sh"
+
+        first_gh_lineno, _ = gh_calls[0]
+        assert export_idx + 1 < first_gh_lineno, (
```
Comment (lugassawan) [inline tests/test_release_sh_gh_repo.py]: **Severity: Medium — Off-by-one risk in comparison (asymmetric index/line-number mixing)**

`export_idx` is a 0-based index; `export_idx + 1` converts it to a 1-based line number. `first_gh_lineno` is already 1-based (from `_gh_call_lines`). The assertion `export_idx + 1 < first_gh_lineno` happens to be correct, but a future reader interpreting `export_idx + 1` as still-index arithmetic could introduce a real off-by-one.

Suggested fix — make both sides unambiguously 1-based:
```python
export_lineno = export_idx + 1
assert export_lineno < first_gh_lineno, (
    f"export GH_REPO (line {export_lineno}) must come before "
    f"first 'gh pr/repo' call (line {first_gh_lineno})"
)
```

File: tests/test_release_sh_gh_repo.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        ORIGIN_URL=$(git remote get-url origin 2>/dev/null || true)
+        GH_REPO=$(printf '%s\\n' "$ORIGIN_URL" \\
+          | sed -E 's#^git@github\\.com:#https://github.com/#' \\
+          | sed -E 's#^https://github\\.com/##; s#\\.git$##')
+        printf '%s\\n' "$GH_REPO"
+    """)
+
+    def _run(self, fake_url: str) -> str:
+        """Stub `git` on PATH to return fake_url, then run the extraction snippet."""
+        stub_dir = Path(os.environ.get("TMPDIR", "/tmp")) / "gh_repo_stub"
```
Comment (lugassawan) [inline tests/test_release_sh_gh_repo.py]: **Severity: Low — Shared stub directory not cleaned up between test runs**

`stub_dir` uses a fixed path under `TMPDIR`. In parallel pytest runs two tests could race on writing `stub_dir/git`, and a stale stub from a prior run could silently affect results.

Suggested fix — use `tempfile.mkdtemp()` with teardown, or the pytest `tmp_path` fixture:
```python
def _run(self, fake_url: str, tmp_path: Path) -> str:
    stub_dir = tmp_path / "git_stub"
    stub_dir.mkdir()
    ...
```
(Propagate `tmp_path` from the test methods.)

**Review by lugassawan — COMMENTED** [review]
## Review Summary

PR #260 makes a targeted, well-scoped fix: exporting `GH_REPO` derived from the `origin` remote URL early in `release.sh` so all subsequent `gh` calls honour it without individual `--repo` flags. The sed-pipeline normalisation covers both SSH and HTTPS URL forms, the validation regex acts as a correctness gate, and the replacement of the final `gh repo view` call with the already-computed variable is a clean simplification. The regression test suite is thorough for the happy-path URL forms. Two issues need attention: a shell injection vector in the test stub and an asymmetric index/line-number naming issue inside `TestExportOrdering`.

Detailed feedback in inline comments.

**Review Decision: COMMENT**

---
_Reviewed by `reviewer` ([swe-workbench](https://github.com/lugassawan/swe-workbench)). Posted 3 inline comments, deduped 0._

File: tests/test_release_sh_gh_repo.py (hunk 1 above)
Comment (lugassawan) [reply tests/test_release_sh_gh_repo.py]: Addressed in 66d28adcc46b19dff74c1f0233bb4b05a036e554: replaced bare f-string interpolation with `shlex.quote(fake_url)` and `printf '%s\n'` in the git stub to eliminate the shell injection vector.

File: tests/test_release_sh_gh_repo.py (hunk 2 above)
Comment (lugassawan) [reply tests/test_release_sh_gh_repo.py]: Addressed in 66d28adcc46b19dff74c1f0233bb4b05a036e554: extracted `export_lineno = export_idx + 1` so both sides of the assertion are unambiguously 1-based line numbers.

File: tests/test_release_sh_gh_repo.py (hunk 3 above)
Comment (lugassawan) [reply tests/test_release_sh_gh_repo.py]: Addressed in 66d28adcc46b19dff74c1f0233bb4b05a036e554: replaced the fixed `gh_repo_stub` path with `tempfile.mkdtemp(prefix='gh_repo_stub_')` and a `try/finally shutil.rmtree` teardown for isolation between runs.

File: tests/test_release_sh_gh_repo.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        export_idx = next(
+            (i for i, ln in enumerate(lines) if "export GH_REPO" in ln),
+            None,
+        )
+        assert export_idx is not None, "export GH_REPO not found in release.sh"
+
+        orphans = [
+            (lineno, ln.strip())
+            for lineno, ln in _gh_call_lines(lines)
+            if lineno <= export_idx + 1
```
Comment (lugassawan) [inline tests/test_release_sh_gh_repo.py]: **Severity: Medium — Boundary condition uses `<=` instead of `<` in `TestNoOrphanCalls`**

```python
if lineno <= export_idx + 1
```

`export_idx + 1` is the 1-based line number of the `export GH_REPO` line itself. The intent is "calls that appear *before* the export" but `<=` includes the export line — making a `gh pr/repo` call on the exact same source line as `export GH_REPO` (impossible in practice, but semantically wrong). This is the same index/lineno mixing fixed in `TestExportOrdering` in the previous round, but not propagated here.

Suggested fix:
```python
export_lineno = export_idx + 1
orphans = [
    (lineno, ln.strip())
    for lineno, ln in _gh_call_lines(lines)
    if lineno < export_lineno   # strictly before the export line
]
```

File: tests/test_release_sh_gh_repo.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            result = subprocess.run(
+                ["bash", "-c", self._SNIPPET],
+                capture_output=True,
+                text=True,
+                env=env,
+            )
+            assert result.returncode == 0, f"snippet failed: {result.stderr}"
+            return result.stdout.strip()
+        finally:
+            import shutil
```
Comment (lugassawan) [inline tests/test_release_sh_gh_repo.py]: **Severity: Low — `shutil` imported inside `finally` block; should be at module top**

```python
finally:
    import shutil
    shutil.rmtree(stub_dir, ignore_errors=True)
```

`shutil` is a stdlib module and won't fail to import, but placing it here deviates from the file's own import-at-top discipline and will be flagged by `isort`/`ruff`. Move it to the module-level imports alongside `os`, `re`, `shlex`, `subprocess`, `tempfile`, and `textwrap`.

**Review by lugassawan — COMMENTED** [review]
## Review Summary

All three previously reported issues have been fully addressed: the shell injection (High) is closed via `shlex.quote` + `printf`, the naming inconsistency (Medium) is resolved with the `export_lineno` variable in `TestExportOrdering`, and the shared temp dir (Low) is replaced with `tempfile.mkdtemp` inside a `try/finally/shutil.rmtree`. No new issues in `scripts/release.sh`. Two small issues remain in the test file: a `<=` vs `<` boundary condition in `TestNoOrphanCalls` (the same index/lineno confusion fixed in `TestExportOrdering` but not propagated), and `shutil` imported inside the `finally` block rather than at module top.

Detailed feedback in inline comments.

**Review Decision: COMMENT**

---
_Re-reviewed by `reviewer` ([swe-workbench](https://github.com/lugassawan/swe-workbench)). Posted 2 inline comments, deduped 0._

File: tests/test_release_sh_gh_repo.py (hunk 4 above)
Comment (lugassawan) [reply tests/test_release_sh_gh_repo.py]: Addressed in 615e4a46e251fa57832897ad768c3003d09e212a: extracted `export_lineno = export_idx + 1` and changed `<=` to `<` so the predicate is strictly-before, matching `TestExportOrdering`'s convention.

File: tests/test_release_sh_gh_repo.py (hunk 5 above)
Comment (lugassawan) [reply tests/test_release_sh_gh_repo.py]: Addressed in 615e4a46e251fa57832897ad768c3003d09e212a: moved `import shutil` to module-level alongside the other stdlib imports.

**Review by lugassawan — COMMENTED** [review]
## Review Summary

All five findings from prior review rounds are cleanly resolved in the final state: the shell injection was eliminated via `shlex.quote` + `printf`; `export_lineno` is now a named variable in both static test classes; each `_run()` invocation gets its own isolated `mkdtemp` directory cleaned up by `try/finally`; the orphan boundary uses strict `<`; and `import shutil` sits at module top. The `GH_REPO` derivation in `scripts/release.sh` is placed correctly before all `gh pr`/`gh repo` action calls. No new issues were introduced across the three commits.

No new findings — ready to merge.

**Review Decision: APPROVE**

---
_Re-reviewed by `reviewer` ([swe-workbench](https://github.com/lugassawan/swe-workbench)). Posted 0 inline comments, deduped 0. **Reviewer verdict: APPROVE** (submitted as COMMENT — GitHub blocks self-approval)._
